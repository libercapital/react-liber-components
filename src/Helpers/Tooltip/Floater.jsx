import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import { Float, Arrow } from './styles';

class Floater extends React.Component {
  handleBlur = () => {
    this.setState({ visible: false });
  };

  handleFocus = () => {
    this.setState({ visible: !this.state.visible });
  };

  isContentPositionUnset = () => {
    const { contentPosition } = this.props;
    return contentPosition === 'unset';
  };

  render = () => {
    const {
      children,
      label,
      show,
      position,
      mousePadding,
      mount,
      contentPosition,
    } = this.props;

    return ReactDOM.createPortal(
      <Float
        visible={show}
        position={
          this.isContentPositionUnset() ? `mouse-${mount}` : `content-${mount}`
        }
        x={`${position.x ? position.x + window.scrollX : 0}px`}
        y={`${position.y ? position.y + window.scrollY : 0}px`}
        mousePadding={mousePadding > 13 ? mousePadding : 13}
        hoverable={show}
        contentPosition={contentPosition}
      >
        <Arrow mount={mount} contentPosition={contentPosition} />
        {label.length > 0 ? <h6>{label}</h6> : null}
        <div>{children}</div>
      </Float>,
      document.getElementById('tooltip'),
    );
  };
}

Floater.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.string,
  ]).isRequired,
  label: PropTypes.string,
  show: PropTypes.bool,
  position: PropTypes.shape({
    x: PropTypes.number,
    y: PropTypes.number,
  }),
  mousePadding: PropTypes.number,
  mount: PropTypes.oneOf(['top', 'right', 'left', 'bottom']),
  contentPosition: PropTypes.oneOf(['start', 'middle', 'end', 'unset']),
};

Floater.defaultProps = {
  label: '',
  show: true,
  position: {},
  mousePadding: 40,
  mount: 'right',
  contentPosition: 'unset',
};

export default Floater;
