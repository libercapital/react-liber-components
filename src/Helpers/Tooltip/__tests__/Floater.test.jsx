import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import Floater from '../Floater';

describe('Test for Floater Component', () => {
  configure({ adapter: new Adapter() });

  const modalRoot = global.document.createElement('div');
  modalRoot.setAttribute('id', 'tooltip');
  const body = global.document.querySelector('body');
  body.appendChild(modalRoot);

  it('should not render without children', () => {
    const error = sinon.stub(console, 'error');
    mount(<Floater />);
    expect(error.calledOnce).toBeTruthy();
    error.restore();
  });

  it('should render correctly with children', () => {
    const wrapper = mount(
      <Floater>
        <p>Hello.</p>
      </Floater>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should spread props correctly', () => {
    const wrapper = mount(
      <Floater label="Helper" position={{ x: 0, y: 0 }} mount="right">
        <p>Hello.</p>
      </Floater>,
    );
    expect(wrapper.find('h6').prop('children')).toEqual('Helper');
    expect(wrapper.find('styles__Float').prop('position')).toEqual(
      'mouse-right',
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with children visible', () => {
    const wrapper = mount(
      <Floater show>
        <p>Hello.</p>
      </Floater>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should spread props correctly visible', () => {
    const wrapper = mount(
      <Floater show label="Helper" position={{ x: 0, y: 0 }} mount="right">
        <p>Hello.</p>
      </Floater>,
    );
    expect(wrapper.find('h6').prop('children')).toEqual('Helper');
    expect(wrapper.find('styles__Float').prop('position')).toEqual(
      'mouse-right',
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with contentPosition ', () => {
    const wrapper = mount(
      <Floater contentPosition="start" show>
        <p>Hello.</p>
      </Floater>,
    );

    expect(wrapper.find('styles__Float').prop('position')).toEqual(
      'content-right',
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
