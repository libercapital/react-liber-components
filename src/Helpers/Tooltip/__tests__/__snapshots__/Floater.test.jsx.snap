// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Floater Component should render correctly with children 1`] = `
<Floater
  contentPosition="unset"
  label=""
  mount="right"
  mousePadding={40}
  position={Object {}}
  show={true}
>
  <Portal
    containerInfo={
      <div
        id="tooltip"
      >
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <div />
        </div>
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
      </div>
    }
  >
    <styles__Float
      contentPosition="unset"
      hoverable={true}
      mousePadding={40}
      position="mouse-right"
      visible={true}
      x="0px"
      y="0px"
    >
      <StyledComponent
        contentPosition="unset"
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [
              [Function],
            ],
            "componentStyle": ComponentStyle {
              "componentId": "styles__Float-sc-1oomni5-2",
              "isStatic": false,
              "lastClassName": "geNPiI",
              "rules": Array [
                "position:absolute;border-radius:4px;background-color:",
                [Function],
                ";box-shadow:0 2px 6px 0 rgba(0,0,0,0.2);z-index:9999;display:block;pointer-events:",
                [Function],
                ";opacity:",
                [Function],
                ";transition:",
                [Function],
                ";color:",
                [Function],
                ";font-size:",
                [Function],
                ";font-weight:",
                [Function],
                ";line-height:20px;",
                [Function],
                " & *{pointer-events:none;}& > h6{font-weight:bold;margin:0px;opacity:0.75;font-family:inherit;font-size:14px;text-align:left;color:",
                [Function],
                ";background-color:",
                [Function],
                ";padding:6px 0px;text-align:center;user-select:none;}& > div{display:flex;width:auto;justify-content:flex-start;align-items:center;padding:12px;overflow:hidden;}",
              ],
            },
            "displayName": "styles__Float",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "styles__Float-sc-1oomni5-2",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        hoverable={true}
        mousePadding={40}
        position="mouse-right"
        visible={true}
        x="0px"
        y="0px"
      >
        <div
          className="styles__Float-sc-1oomni5-2 geNPiI"
          style={
            Object {
              "left": "0px",
              "top": "0px",
            }
          }
          x="0px"
          y="0px"
        >
          <styles__Arrow
            contentPosition="unset"
            mount="right"
          >
            <StyledComponent
              contentPosition="unset"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styles__Arrow-sc-1oomni5-1",
                    "isStatic": false,
                    "lastClassName": "hwShoT",
                    "rules": Array [
                      "position:absolute;",
                      [Function],
                      " overflow:hidden;pointer-events:none;padding:8px;box-sizing:border-box;&::after{content:'';position:absolute;pointer-events:none;width:15px;height:15px;background:#fff;transform:rotate(45deg);",
                      [Function],
                      " box-shadow:0 2px 6px 0 rgba(0,0,0,0.2);}",
                    ],
                  },
                  "displayName": "styles__Arrow",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styles__Arrow-sc-1oomni5-1",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              mount="right"
            >
              <div
                className="styles__Arrow-sc-1oomni5-1 hwShoT"
              />
            </StyledComponent>
          </styles__Arrow>
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
      </StyledComponent>
    </styles__Float>
  </Portal>
</Floater>
`;

exports[`Test for Floater Component should render correctly with children visible 1`] = `
<Floater
  contentPosition="unset"
  label=""
  mount="right"
  mousePadding={40}
  position={Object {}}
  show={true}
>
  <Portal
    containerInfo={
      <div
        id="tooltip"
      >
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <div />
        </div>
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <h6>
            Helper
          </h6>
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
      </div>
    }
  >
    <styles__Float
      contentPosition="unset"
      hoverable={true}
      mousePadding={40}
      position="mouse-right"
      visible={true}
      x="0px"
      y="0px"
    >
      <StyledComponent
        contentPosition="unset"
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [
              [Function],
            ],
            "componentStyle": ComponentStyle {
              "componentId": "styles__Float-sc-1oomni5-2",
              "isStatic": false,
              "lastClassName": "geNPiI",
              "rules": Array [
                "position:absolute;border-radius:4px;background-color:",
                [Function],
                ";box-shadow:0 2px 6px 0 rgba(0,0,0,0.2);z-index:9999;display:block;pointer-events:",
                [Function],
                ";opacity:",
                [Function],
                ";transition:",
                [Function],
                ";color:",
                [Function],
                ";font-size:",
                [Function],
                ";font-weight:",
                [Function],
                ";line-height:20px;",
                [Function],
                " & *{pointer-events:none;}& > h6{font-weight:bold;margin:0px;opacity:0.75;font-family:inherit;font-size:14px;text-align:left;color:",
                [Function],
                ";background-color:",
                [Function],
                ";padding:6px 0px;text-align:center;user-select:none;}& > div{display:flex;width:auto;justify-content:flex-start;align-items:center;padding:12px;overflow:hidden;}",
              ],
            },
            "displayName": "styles__Float",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "styles__Float-sc-1oomni5-2",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        hoverable={true}
        mousePadding={40}
        position="mouse-right"
        visible={true}
        x="0px"
        y="0px"
      >
        <div
          className="styles__Float-sc-1oomni5-2 geNPiI"
          style={
            Object {
              "left": "0px",
              "top": "0px",
            }
          }
          x="0px"
          y="0px"
        >
          <styles__Arrow
            contentPosition="unset"
            mount="right"
          >
            <StyledComponent
              contentPosition="unset"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styles__Arrow-sc-1oomni5-1",
                    "isStatic": false,
                    "lastClassName": "hwShoT",
                    "rules": Array [
                      "position:absolute;",
                      [Function],
                      " overflow:hidden;pointer-events:none;padding:8px;box-sizing:border-box;&::after{content:'';position:absolute;pointer-events:none;width:15px;height:15px;background:#fff;transform:rotate(45deg);",
                      [Function],
                      " box-shadow:0 2px 6px 0 rgba(0,0,0,0.2);}",
                    ],
                  },
                  "displayName": "styles__Arrow",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styles__Arrow-sc-1oomni5-1",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              mount="right"
            >
              <div
                className="styles__Arrow-sc-1oomni5-1 hwShoT"
              />
            </StyledComponent>
          </styles__Arrow>
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
      </StyledComponent>
    </styles__Float>
  </Portal>
</Floater>
`;

exports[`Test for Floater Component should render correctly with contentPosition  1`] = `
<Floater
  contentPosition="start"
  label=""
  mount="right"
  mousePadding={40}
  position={Object {}}
  show={true}
>
  <Portal
    containerInfo={
      <div
        id="tooltip"
      >
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <div />
        </div>
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <h6>
            Helper
          </h6>
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <h6>
            Helper
          </h6>
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
        <div
          class="styles__Float-sc-1oomni5-2 cIuPTt"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 kcnzTJ"
          />
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
      </div>
    }
  >
    <styles__Float
      contentPosition="start"
      hoverable={true}
      mousePadding={40}
      position="content-right"
      visible={true}
      x="0px"
      y="0px"
    >
      <StyledComponent
        contentPosition="start"
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [
              [Function],
            ],
            "componentStyle": ComponentStyle {
              "componentId": "styles__Float-sc-1oomni5-2",
              "isStatic": false,
              "lastClassName": "cIuPTt",
              "rules": Array [
                "position:absolute;border-radius:4px;background-color:",
                [Function],
                ";box-shadow:0 2px 6px 0 rgba(0,0,0,0.2);z-index:9999;display:block;pointer-events:",
                [Function],
                ";opacity:",
                [Function],
                ";transition:",
                [Function],
                ";color:",
                [Function],
                ";font-size:",
                [Function],
                ";font-weight:",
                [Function],
                ";line-height:20px;",
                [Function],
                " & *{pointer-events:none;}& > h6{font-weight:bold;margin:0px;opacity:0.75;font-family:inherit;font-size:14px;text-align:left;color:",
                [Function],
                ";background-color:",
                [Function],
                ";padding:6px 0px;text-align:center;user-select:none;}& > div{display:flex;width:auto;justify-content:flex-start;align-items:center;padding:12px;overflow:hidden;}",
              ],
            },
            "displayName": "styles__Float",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "styles__Float-sc-1oomni5-2",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        hoverable={true}
        mousePadding={40}
        position="content-right"
        visible={true}
        x="0px"
        y="0px"
      >
        <div
          className="styles__Float-sc-1oomni5-2 cIuPTt"
          style={
            Object {
              "left": "0px",
              "top": "0px",
            }
          }
          x="0px"
          y="0px"
        >
          <styles__Arrow
            contentPosition="start"
            mount="right"
          >
            <StyledComponent
              contentPosition="start"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styles__Arrow-sc-1oomni5-1",
                    "isStatic": false,
                    "lastClassName": "kcnzTJ",
                    "rules": Array [
                      "position:absolute;",
                      [Function],
                      " overflow:hidden;pointer-events:none;padding:8px;box-sizing:border-box;&::after{content:'';position:absolute;pointer-events:none;width:15px;height:15px;background:#fff;transform:rotate(45deg);",
                      [Function],
                      " box-shadow:0 2px 6px 0 rgba(0,0,0,0.2);}",
                    ],
                  },
                  "displayName": "styles__Arrow",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styles__Arrow-sc-1oomni5-1",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              mount="right"
            >
              <div
                className="styles__Arrow-sc-1oomni5-1 kcnzTJ"
              />
            </StyledComponent>
          </styles__Arrow>
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
      </StyledComponent>
    </styles__Float>
  </Portal>
</Floater>
`;

exports[`Test for Floater Component should spread props correctly 1`] = `
<Floater
  contentPosition="unset"
  label="Helper"
  mount="right"
  mousePadding={40}
  position={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  show={true}
>
  <Portal
    containerInfo={
      <div
        id="tooltip"
      >
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <div />
        </div>
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <h6>
            Helper
          </h6>
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
      </div>
    }
  >
    <styles__Float
      contentPosition="unset"
      hoverable={true}
      mousePadding={40}
      position="mouse-right"
      visible={true}
      x="0px"
      y="0px"
    >
      <StyledComponent
        contentPosition="unset"
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [
              [Function],
            ],
            "componentStyle": ComponentStyle {
              "componentId": "styles__Float-sc-1oomni5-2",
              "isStatic": false,
              "lastClassName": "geNPiI",
              "rules": Array [
                "position:absolute;border-radius:4px;background-color:",
                [Function],
                ";box-shadow:0 2px 6px 0 rgba(0,0,0,0.2);z-index:9999;display:block;pointer-events:",
                [Function],
                ";opacity:",
                [Function],
                ";transition:",
                [Function],
                ";color:",
                [Function],
                ";font-size:",
                [Function],
                ";font-weight:",
                [Function],
                ";line-height:20px;",
                [Function],
                " & *{pointer-events:none;}& > h6{font-weight:bold;margin:0px;opacity:0.75;font-family:inherit;font-size:14px;text-align:left;color:",
                [Function],
                ";background-color:",
                [Function],
                ";padding:6px 0px;text-align:center;user-select:none;}& > div{display:flex;width:auto;justify-content:flex-start;align-items:center;padding:12px;overflow:hidden;}",
              ],
            },
            "displayName": "styles__Float",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "styles__Float-sc-1oomni5-2",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        hoverable={true}
        mousePadding={40}
        position="mouse-right"
        visible={true}
        x="0px"
        y="0px"
      >
        <div
          className="styles__Float-sc-1oomni5-2 geNPiI"
          style={
            Object {
              "left": "0px",
              "top": "0px",
            }
          }
          x="0px"
          y="0px"
        >
          <styles__Arrow
            contentPosition="unset"
            mount="right"
          >
            <StyledComponent
              contentPosition="unset"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styles__Arrow-sc-1oomni5-1",
                    "isStatic": false,
                    "lastClassName": "hwShoT",
                    "rules": Array [
                      "position:absolute;",
                      [Function],
                      " overflow:hidden;pointer-events:none;padding:8px;box-sizing:border-box;&::after{content:'';position:absolute;pointer-events:none;width:15px;height:15px;background:#fff;transform:rotate(45deg);",
                      [Function],
                      " box-shadow:0 2px 6px 0 rgba(0,0,0,0.2);}",
                    ],
                  },
                  "displayName": "styles__Arrow",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styles__Arrow-sc-1oomni5-1",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              mount="right"
            >
              <div
                className="styles__Arrow-sc-1oomni5-1 hwShoT"
              />
            </StyledComponent>
          </styles__Arrow>
          <h6>
            Helper
          </h6>
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
      </StyledComponent>
    </styles__Float>
  </Portal>
</Floater>
`;

exports[`Test for Floater Component should spread props correctly visible 1`] = `
<Floater
  contentPosition="unset"
  label="Helper"
  mount="right"
  mousePadding={40}
  position={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  show={true}
>
  <Portal
    containerInfo={
      <div
        id="tooltip"
      >
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <div />
        </div>
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <h6>
            Helper
          </h6>
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
        <div
          class="styles__Float-sc-1oomni5-2 geNPiI"
          style="left: 0px; top: 0px;"
          x="0px"
          y="0px"
        >
          <div
            class="styles__Arrow-sc-1oomni5-1 hwShoT"
          />
          <h6>
            Helper
          </h6>
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
      </div>
    }
  >
    <styles__Float
      contentPosition="unset"
      hoverable={true}
      mousePadding={40}
      position="mouse-right"
      visible={true}
      x="0px"
      y="0px"
    >
      <StyledComponent
        contentPosition="unset"
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [
              [Function],
            ],
            "componentStyle": ComponentStyle {
              "componentId": "styles__Float-sc-1oomni5-2",
              "isStatic": false,
              "lastClassName": "geNPiI",
              "rules": Array [
                "position:absolute;border-radius:4px;background-color:",
                [Function],
                ";box-shadow:0 2px 6px 0 rgba(0,0,0,0.2);z-index:9999;display:block;pointer-events:",
                [Function],
                ";opacity:",
                [Function],
                ";transition:",
                [Function],
                ";color:",
                [Function],
                ";font-size:",
                [Function],
                ";font-weight:",
                [Function],
                ";line-height:20px;",
                [Function],
                " & *{pointer-events:none;}& > h6{font-weight:bold;margin:0px;opacity:0.75;font-family:inherit;font-size:14px;text-align:left;color:",
                [Function],
                ";background-color:",
                [Function],
                ";padding:6px 0px;text-align:center;user-select:none;}& > div{display:flex;width:auto;justify-content:flex-start;align-items:center;padding:12px;overflow:hidden;}",
              ],
            },
            "displayName": "styles__Float",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "styles__Float-sc-1oomni5-2",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        hoverable={true}
        mousePadding={40}
        position="mouse-right"
        visible={true}
        x="0px"
        y="0px"
      >
        <div
          className="styles__Float-sc-1oomni5-2 geNPiI"
          style={
            Object {
              "left": "0px",
              "top": "0px",
            }
          }
          x="0px"
          y="0px"
        >
          <styles__Arrow
            contentPosition="unset"
            mount="right"
          >
            <StyledComponent
              contentPosition="unset"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "styles__Arrow-sc-1oomni5-1",
                    "isStatic": false,
                    "lastClassName": "hwShoT",
                    "rules": Array [
                      "position:absolute;",
                      [Function],
                      " overflow:hidden;pointer-events:none;padding:8px;box-sizing:border-box;&::after{content:'';position:absolute;pointer-events:none;width:15px;height:15px;background:#fff;transform:rotate(45deg);",
                      [Function],
                      " box-shadow:0 2px 6px 0 rgba(0,0,0,0.2);}",
                    ],
                  },
                  "displayName": "styles__Arrow",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "styles__Arrow-sc-1oomni5-1",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              mount="right"
            >
              <div
                className="styles__Arrow-sc-1oomni5-1 hwShoT"
              />
            </StyledComponent>
          </styles__Arrow>
          <h6>
            Helper
          </h6>
          <div>
            <p>
              Hello.
            </p>
          </div>
        </div>
      </StyledComponent>
    </styles__Float>
  </Portal>
</Floater>
`;
